e2e/vaultstatefulset/vaultstatefulset_test.go

```diff
@@ -0,0 +1,296 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+//go:build e2e
+
+package vaultstatefulset
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/edgelesssys/contrast/e2e/internal/contrasttest"
+	"github.com/edgelesssys/contrast/e2e/internal/kubeclient"
+	"github.com/edgelesssys/contrast/internal/kuberesource"
+	"github.com/edgelesssys/contrast/internal/manifest"
+	"github.com/edgelesssys/contrast/internal/platforms"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestVaultStatefulset(t *testing.T) {
+	platform, err := platforms.FromString(contrasttest.Flags.PlatformStr)
+	require.NoError(t, err)
+	ct := contrasttest.New(t)
+
+	runtimeHandler, err := manifest.RuntimeHandler(platform)
+	require.NoError(t, err)
+
+	resources := kuberesource.Vault(ct.Namespace)
+
+	coordinator := kuberesource.CoordinatorBundle()
+
+	resources = append(resources, coordinator...)
+
+	resources = kuberesource.PatchRuntimeHandlers(resources, runtimeHandler)
+
+	resources = kuberesource.AddPortForwarders(resources)
+
+	ct.Init(t, resources)
+
+	require.True(t, t.Run("generate", ct.Generate), "contrast generate needs to succeed for subsequent tests")
+
+	// Overwrite the workloadSecretID in the manifest, to align with the value set in the
+	// unsealing configuration of the Vault deployment.
+	ct.PatchManifest(t, func(m manifest.Manifest) manifest.Manifest {
+		for key, policy := range m.Policies {
+			// TODO only do this for Vault image, not the other deployment
+			policy.WorkloadSecretID = "vault_unsealing"
+			m.Policies[key] = policy
+		}
+		return m
+	})
+
+	require.True(t, t.Run("apply", ct.Apply), "Kubernetes resources need to be applied for subsequent tests")
+
+	require.True(t, t.Run("set", ct.Set), "contrast set needs to succeed for subsequent tests")
+
+	require.True(t, t.Run("contrast verify", ct.Verify), "contrast verify needs to succeed for subsequent tests")
+
+	// Get Vault pod for subsequent tests.
+	ctx, cancel := context.WithTimeout(t.Context(), ct.FactorPlatformTimeout(2*60*time.Second))
+	defer cancel()
+	require.NoError(t, ct.Kubeclient.WaitFor(ctx, kubeclient.Ready, kubeclient.StatefulSet{}, ct.Namespace, "vault"))
+	pods, err := ct.Kubeclient.PodsFromOwner(ctx, ct.Namespace, "StatefulSet", "vault")
+	require.NoError(t, err)
+	require.Len(t, pods, 1)
+
+	var token string
+	loggingPath := "/vault/data/openbao.log"
+
+	t.Run("transit engine api auto-unseals Vault", func(t *testing.T) {
+		require := require.New(t)
+		ctx, cancel := context.WithTimeout(t.Context(), ct.FactorPlatformTimeout(2*60*time.Second))
+		defer cancel()
+
+		vaultLogs, stdErr, err := ct.Kubeclient.ExecContainer(ctx, ct.Namespace, pods[0].Name, "openbao-server", []string{"cat", loggingPath})
+		require.NoError(err, "stdout: %s, stderr: %s", vaultLogs, stdErr)
+
+		coordinatorLogs, err := ct.Kubeclient.GetContainerLogs(ctx, ct.Namespace, "StatefulSet", "coordinator", "coordinator")
+		require.NoError(err)
```

Comment:
I don't think we should be looking at the Coordinator logs here, but rather focus on the result: does the Vault eventually become unsealed? It should be possible to use https://developer.hashicorp.com/vault/docs/commands/status or an equivalent HTTP request.